components/tidb_query/src/rpn_expr/impl_json.rs

```diff
@@ -50,25 +50,24 @@ fn json_object_validator(expr: &tipb::Expr) -> Result<()> {
 fn json_object(raw_args: &[ScalarValueRef]) -> Result<Option<Json>> {
     let mut pairs = BTreeMap::new();
     for chunk in raw_args.chunks(2) {
-        // chunk.len() must be 1 or 2 here.
-        let key =
-            Evaluable::borrow_scalar_value_ref(&chunk[0])
-                .as_ref()
-                .map_or_else(
-                    || {
-                        Err(other_err!(
-                            "Data truncation: JSON documents may not contain NULL member names."
-                        ))
-                    },
-                    |v: &Bytes| {
-                        Ok(String::from_utf8(v.to_owned())
-                            .map_err(|e| crate::codec::Error::from(e))?)
-                    },
-                )?;
-
-        let value = Evaluable::borrow_scalar_value_ref(&chunk[1])
-            .as_ref()
-            .map_or(Json::None, |v: &Json| v.to_owned());
+        // chunk.len() must be 2 here.
```

Comment:
Then you can assert it.

Reply:
> Then you can assert it.

Add `assert!(chunk.len() >= 2)` now.

Reply:
The comment can be removed because the assertion has already described the requirement.

Reply:
> The comment can be removed because the assertion has already described the requirement.

Removed now.